### ISSUE TITLE
feat: add profile tags to all profile types

### ISSUE THREAD

**Issue by benfurber** [ONEARMY/community-platform#3908]
Many elements of the Precious Plastic specific profile types (workspaces, machine shop, etc.) are highly specifc. This makes extension for other projects more complex.

The main is to remove all the space fields for the PP specific profile types and replace them with a more generic profile tags field which can then also be added to the member and space profile types.

To-do:
- [x] Add tags to typing
- [x] Add tags to the user update functions
- [x] Add tags to machine shops profiletype setting
- [x] Add tags to machine shops profile page
- [x] Update machine shop data on prod DB 
- [x] Add tags to workspaces profiletype setting
- [x] Add tags to workspaces shops profile page
- [x] Update workspaces data on prod DB 
- [x] Add tags to collection point profiletype setting
- [x] Add tags to collection point shops profile page
- [x] Update collection point data on prod DB 
- [x] Add tags to community profiletype setting
- [x] Add tags to community profile page
- [x] Add tags to spaced profiletype setting
- [x] Add tags to spaced profile page
- [x] Update spaces tags data to theme level
- [x] Update member tags data to theme level
- [x] Add tags to member profiletype setting
- [x] Add tags to member profile page
- [ ] Update design of select list field (for this field and other tag fields)
- [x] Add rule so that five really is the max number that can be selected

**Comment by benfurber** [comment]
@davehakkens Do you mean have a seperate list of tag options?

**Comment by davehakkens** [comment]
> @davehakkens Do you mean have a seperate list of tag options?

Could be. But not sure if that is worth the added complexity at this point? What do you think?

I do know that at minimum we needs tags for spaces

**Comment by benfurber** [comment]
@davehakkens As community and members don't have any custom fields atm it's super easy to add them.

Can you please provide a list of labels for:
1. PP members, PK members and FF members.
2. PP community.
3. PK space and FF space.

Happy receive 1, 2 and 3 whenever. Need all parts of 1 or 3 at the same time to make releasing simple.

**Comment by benfurber** [comment]
> > @davehakkens Do you mean have a seperate list of tag options?
> 
> Could be. But not sure if that is worth the added complexity at this point? What do you think?
> 
> I do know that at minimum we needs tags for spaces
> 
> Members is a nice to have

Sorry, @davehakkens I missed that the first setence was a question. I think it's worth doing all of this. I see the value from a product prospect, it's not that much more work to do it all and then the code is simplier at the end as we have one way of doing it all.

**Comment by davehakkens** [comment]
@benfurber Why is PP Community (2) different from PK and FF Space (3), it would eventually all work the same right?
-> All spaces on an instance use the same profile-tags

**Comment by benfurber** [comment]
@davehakkens Yes, all ProfileTypes on an instance use the same profile tags. But there are different tags for PK, PP and FF right?

For 2 I mean the PP "COMMUNITY_BUILDER" profile type, it's coded differently to "SPACE". Should they be the same thing?

**Comment by davehakkens** [comment]
> @davehakkens Yes, all ProfileTypes on an instance use the same profile tags. But there are different tags for PK, PP and FF right?

Correct.

> For 2 I mean the PP "COMMUNITY_BUILDER" profile type, it's coded differently to "SPACE". Should they be the same thing?

Strange, what is different between them? In my head that one was always the same as the others.



Regarding the tag list. Here is a [document](https://docs.google.com/document/d/1_DqCvDtDJn2CBAfi4D67Tdd5hua0JmBvuOv-Fg8FBmM/edit?tab=t.0)
It's not 100% thought out, as we would still need to do this work for PK somewhere in the winter. 
But perhaps we could use it as a start and iterate as we go?

**Comment by benfurber** [comment]
Re. COMMUNITY_BUILDER versus SPACE. I'll remove the difference.

Thanks for the doc. Super easy to change/extend in the future. Once the DB migration work is done, settings the tags will move to the DB anyway.
